helm/ako/templates/statefulset.yaml

```diff
@@ -40,6 +40,10 @@ spec:
             {{- toYaml .Values.securityContext | nindent 12 }}
           image: "{{ .Values.image.repository }}:{{ .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
+          lifecycle:
+            preStop:
+              exec:
+                command: ["/bin/sh","-c","if [ $USE_PVC ]; then cd $LOG_FILE_PATH; old_log_tar_file=$LOG_FILE_PATH/old_logs.tar ;old_pods_gz_file=$LOG_FILE_PATH/old_logs.tar.gz; if [ ! -f \"$old_pods_gz_file\" ] ; then tar -cf $old_log_tar_file $POD_NAME*.log* ; else gunzip $old_pods_gz_file;tar rf $old_log_tar_file $POD_NAME*.log*;fi; rm -rf $POD_NAME*.log*; gzip $old_log_tar_file; fi;"]
```

Comment:
Would it be better if we ad this in a script inside the docker file and call that script ?
The script is bot long, so  it would be easier to understand and later we can add to that script if required. 

@sudswasavi  what do you think ?

Reply:
@monotosh-avi: If I understand correctly, you are suggesting: check-in shell script in AKO git repo.. Write copy statement in docker file to cp that file inside container at specific path (say at /var/script.sh) while building ako image and then call that script from pre-stop hook.  Sounds good. 
Only thing is - any changes in code needs to be done by logging into docker instead of 'kubectl edit' command.

Reply:
yes @akshayhavile, we can add the script in the ako container image.
We should be able to catch any issue in our testing and I think customer would't have to modify this script.

Reply:
A script:
  - is good for debuggability.
  - is baked into the image so can't be changed by the customer.

Unless we have any side effects of using the scripts, that will check through our course of system test,